packages/babel-helper-create-class-features-plugin/src/misc.js

```diff
@@ -25,26 +25,24 @@ const referenceVisitor = {
     }
   },
 };
+function handleClassTDZ(path, state) {
+  if (
+    state.classBinding &&
+    state.classBinding === path.scope.getBinding(path.node.name)
+  ) {
+    const classNameTDZError = state.file.addHelper("classNameTDZError");
+    const throwNode = t.callExpression(classNameTDZError, [
+      t.stringLiteral(path.node.name),
+    ]);
+
+    path.replaceWith(t.sequenceExpression([throwNode, path.node]));
+    path.skip();
+  }
+}
 
-const classFieldDefinitionEvaluationTDZVisitor = traverse.visitors.merge([
-  {
-    ReferencedIdentifier(path) {
-      if (
-        this.classBinding &&
-        this.classBinding === path.scope.getBinding(path.node.name)
-      ) {
-        const classNameTDZError = this.file.addHelper("classNameTDZError");
-        const throwNode = t.callExpression(classNameTDZError, [
-          t.stringLiteral(path.node.name),
-        ]);
-
-        path.replaceWith(t.sequenceExpression([throwNode, path.node]));
-        path.skip();
-      }
-    },
-  },
-  environmentVisitor,
```

Comment:
Context for reviewers: `environmentVisitor` wasn't actually used. It defines a visitor for `Method` so that it can skip all its children except for the computed key, but since we were calling `methodPath.traverse` that visitor was never called.

Reply:
@nicolo-ribaudo Should I add this as a comment in the code as well? Can help future contributors as a reference.

Reply:
Maybe we could add a comment in the file where `environmentVisitor` is defined, explained that it should be used when traversing the whole class and not for specific class elements.

Reply:
@nicolo-ribaudo Done. PTAL.